src/lib/y2storage/clients/partitions_proposal.rb

```diff
@@ -0,0 +1,140 @@
+#!/usr/bin/env ruby
+#
+# encoding: utf-8
+
+# Copyright (c) [2017] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "y2storage"
+require "y2storage/actions_presenter"
+require "y2storage/dialogs/proposal"
+require "installation/proposal_client"
+
+Yast.import "Popup"
+
+module Y2Storage
+  module Clients
+    # Proposal client to show the list of storage actions.
+    #
+    # This client is used in autoyast summary dialog, where a new
+    # client instance is created with each event that requires to
+    # update the summary dialog content.
+    #
+    # To manage collapsing/expanding subvolumes list it is necessary
+    # to save the previous state of that list, but it is not possible
+    # to do that in the client instance itself because a new client
+    # instance is created each time.
+    #
+    # To solve that, instance class attributes are used. PartitionsProposal
+    # class has an internal state that is updated with each object creation.
+    #
+    # @see PartitionsProposal.update_state
+    class PartitionsProposal < ::Installation::ProposalClient
+      include Yast::Logger
+
+      def initialize
+        if storage_manager.proposal.nil?
```

Comment:
This implies displaying this proposal always destroys the result of the expert partitioner, so expert partitioner simply cannot be used. Isn't it?

Take a look to how this is solved in `inst_disk_proposal` that, as proved by its unit tests, only proposes something when really needed, respecting a manually set staging devicegraph.
https://github.com/yast/yast-storage-ng/blob/master/src/lib/y2storage/clients/inst_disk_proposal.rb#L41